front/components/agent_builder/instructions/AgentBuilderInstructionsBlock.tsx

```diff
@@ -1,10 +1,62 @@
-import { Page } from "@dust-tt/sparkle";
-import React from "react";
+import {
+  ArrowPathIcon,
+  Button,
+  Label,
+  Page,
+  Separator,
+  XMarkIcon,
+} from "@dust-tt/sparkle";
+import React, { useState } from "react";
+import { useFormContext } from "react-hook-form";
 
+import { useAgentBuilderContext } from "@app/components/agent_builder/AgentBuilderContext";
+import type { AgentBuilderFormData } from "@app/components/agent_builder/AgentBuilderFormContext";
 import { AdvancedSettings } from "@app/components/agent_builder/instructions/AdvancedSettings";
 import { AgentBuilderInstructionsEditor } from "@app/components/agent_builder/instructions/AgentBuilderInstructionsEditor";
+import { AgentInstructionsHistory } from "@app/components/agent_builder/instructions/AgentInstructionsHistory";
+import { useAgentConfigurationHistory } from "@app/lib/swr/assistants";
+import type { LightAgentConfigurationType } from "@app/types";
+
+interface AgentBuilderInstructionsBlockProps {
+  agentConfigurationId: string | null;
+}
+
+export function AgentBuilderInstructionsBlock({
+  agentConfigurationId,
+}: AgentBuilderInstructionsBlockProps) {
+  const { owner } = useAgentBuilderContext();
+  const { setValue } = useFormContext<AgentBuilderFormData>();
+  const [compareVersion, setCompareVersion] =
+    useState<LightAgentConfigurationType | null>(null);
+  const [isInstructionDiffMode, setIsInstructionDiffMode] = useState(false);
+
+  const { agentConfigurationHistory } = useAgentConfigurationHistory({
+    workspaceId: owner.sId,
+    agentConfigurationId,
+    disabled: !agentConfigurationId,
+    limit: 30,
+  });
+
+  const dateFormatter = new Intl.DateTimeFormat("en-US", {
+    year: "numeric",
+    month: "2-digit",
+    day: "2-digit",
+    hour: "numeric",
+    minute: "numeric",
+    hour12: true,
+  });
```

Comment:
We have access to [date-fns](https://date-fns.org/v4.1.0/docs/format), easier to format. I think it would be just `format(date, 'Pp')`
P = Localized date format
p = Localized time format
Pp = Auto combination of localized date & time format